packages/react-core/src/components/Label/Label.tsx

```diff
@@ -118,6 +121,25 @@ export const Label: React.FunctionComponent<LabelProps> = ({
     };
   });
 
+  React.useEffect(() => {
+    if (onLabelClick && isOverflowLabel) {
+      // eslint-disable-next-line no-console
+      console.warn(
+        'Overflow labels cannot have onClick passed, this results in invalid HTML. Please remove either the isOverflowLabel or onClick prop.'
+      );
+    } else if (onLabelClick && href) {
```

Comment:
We can remove the first if block/mentions of isOverflowLabel from this useEffect